irc/channel.go

```diff
@@ -539,6 +541,27 @@ func (channel *Channel) ClientPrefixes(client *Client, isMultiPrefix bool) strin
 	}
 }
 
+func (channel *Channel) ClientModeStrings(client *Client) []string {
+	channel.stateMutex.RLock()
+	defer channel.stateMutex.RUnlock()
+	modes, present := channel.members[client]
+	if !present {
+		return []string{}
+	} else {
+		return modes.Strings()
+	}
+}
+
+func (channel *Channel) ClientJoinTime(client *Client) *time.Time {
```

Comment:
[Idiomatic Go](https://golang.org/pkg/time/) is to use the zero time and test `time.IsZero()` instead of using `*time.Time`:

>The zero value of type Time is January 1, year 1, 00:00:00.000000000 UTC. As this time is unlikely to come up in practice, the IsZero method gives a simple way of detecting a time that has not been initialized explicitly.